cfpb/cfgov-refresh
# Optionally enable Wagtail's Postgres full text search support and index PageRevisions

The upgrade to 2.5 removed an obstacle to indexed admin searching because it removed the naming clash between Wagtail and Haystack over the `update_index` command.

This version also makes available Postgres' full-text search as an option (using the "fulltext" backend), but not the default, so that we can do further iteration on Wagtail's search without disrupting current admin and external link search expectations.

This PR also adds a proxy model for `PageRevision` so that it can be searched as well. Because it's a separate Django model, and not a descendent of `Page`,  it will not clutter any `Page` searches, and would have to be explicitly searched.

## How to test this PR

To give it a try, check out the branch, migrate and index:

```
./cfgov/manage.py migrate
./cfgov/manage.py wagtail_update_index
./cfgov/manage.py shell
```

Then perform a querying the "fulltext" backend:

```python
from v1.models import IndexedPageRevision
from wagtail.core.models import Page
from wagtail.search.backends import get_search_backend

search_query = "sandbox"
page_results = Page.objects.search(search_query, backend="fulltext")
revision_results = get_search_backend("fulltext").search(search_query, IndexedPageRevision
```

## Checklist
